lib/salus/scan_report.rb

```diff
@@ -0,0 +1,133 @@
+require 'json'
+
+module Salus
+  class ScanReport
+    attr_reader :scanner_name, :running_time
+
+    def initialize(scanner_name)
+      @scanner_name = scanner_name
+      @passed = nil
+      @running_time = nil
+      @info = {}
+      @errors = []
+    end
+
+    def record
+      started_at = monotime
+
+      yield
+
+      # If the block failed to pass/fail the scan,
+      # default to pass if no errors were recorded
+      @passed = @errors.empty? if @passed.nil?
```

Comment:
This is an interesting idea.

Would this only happen if a developer wrote the scanner class incorrectly? If so I would default to failure so that this bug is more likely to be caught.

Reply:
Actually for the dep reporting scanners I see this is kind of necessary. Happy to keep then.

Reply:
yeah that's exactly why it's here. i would also be okay with requiring scanners to report their status but i'm not that opinionated (and this was faster than that)